### PR TITLE
Add tailwind

### LINKED ISSUE #7891: Add tailwind into our build process
### What we should change and why (this is tech debt)

As discussed in delivery circle, we decide to start with Tailwind and review in 3 months if this was the right way to go.

This issue is here to integrate tailwind into our build.

### Sources
 - https://dev.to/andrewmcodes/webpacker-6-tailwind-css-2-0-integration-3oe6
 - https://gorails.com/episodes/tailwind-css-framework-with-rails
 - https://dev.to/davidteren/getting-started-with-rails-6-1-tailwindcss-jit-webpacker-postcss-8-25g7
 - https://evilmartians.com/chronicles/set-up-tailwind-css-jit-in-a-rails-project-to-compile-styles-20x-faster

### Context
 - #7543
 - #7539
 - #7524
 - https://community.openfoodnetwork.org/t/choosing-our-css-framework/2328
 - https://community.openfoodnetwork.org/t/delivery-circle-meeting-notes-july-6th/2348

### PR DESCRIPTION
**Pull request by jibees**
#### What? Why?

This PR adds tailwindcss as a postcss (7) plugin and integrate it into the webpack list of assets to build.

Closes #7891 

**TODO**:
 - test the build

#### Release notes

Changelog Category: User facing changes | Technical changes

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,19 +12,26 @@
...
     "@storybook/server": "^6.3.4",
+    "autoprefixer": "^9",
     "jasmine-core": "~2.4.1",
     "karma": "~6.3.4",
     "karma-chrome-launcher": "~3.1.0",
     "karma-coffee-preprocessor": "~1.0.1",
     "karma-jasmine": "~0.3.8",
+    "postcss": "^7",
+    "postcss-loader": "~3.0.0",
+    "tailwindcss": "npm:@tailwindcss/postcss7-compat",
```
Comment (Matt-Yorkley) [inline package.json:23]: :+1:

File: package.json [hunk 2]
```diff
@@ -12,19 +12,26 @@
...
     "webpack-dev-server": "^3.11.2"
   },
   "license": "AGPL-3.0",
   "dependencies": {
+    "@rails/activestorage": "^6.1.4",
+    "@rails/ujs": "^6.1.4",
```
Comment (Matt-Yorkley) [inline package.json:29]: We're not currently using ActiveStorage or UJS, so I'm not sure we need these here...

File: package.json [hunk 3]
```diff
@@ -12,19 +12,26 @@
...
     "@rails/webpacker": "5.4.0",
     "flatpickr": "^4.6.9",
     "moment": "^2.29.1",
     "shortcut-buttons-flatpickr": "^0.3.1",
+    "turbolinks": "^5.2.0",
```
[the remark this answers is not in the document]
Comment (jibees) [reply package.json:34]: I followed a tutorial, but you'r right, that's out of scope...

**Comment by Matt-Yorkley** [comment]
I'll try to take a look at this later on but I'm not sure how to test PurgeCSS... :thinking:

**Comment by jibees** [comment]
> I'll try to take a look at this later today, but I'm not sure how to test PurgeCSS... 🤔

Thanks a lot. I succeeded in merging the two webpack config rules, but seems that there is no tailwind at all. 

Regarding the Purge option, the concept is that tailwind should include in the _final_ css file only the classes that are actually used. To test it, simply check that the CSS file contains only the right classes.

File: tailwind.config.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+const plugin = require("tailwindcss/plugin");
+
+module.exports = {
+  purge: [],
```
Comment (Matt-Yorkley) [inline tailwind.config.js:4]: We should be able to add `mode: "jit"` here. And we'll need to configure the purge option, something like:
```
purge: {
    enabled: true,
    content: [
      "./app/views/**/*.html.haml",
      "./app/assets/javascript/templates**/*.js"
    ]
  }
```

File: package.json [hunk 5]
```diff
@@ -11,12 +11,17 @@
...
     "@storybook/addon-docs": "^6.3.7",
+    "@storybook/addon-postcss": "^2.0.0",
     "@storybook/server": "^6.3.7",
+    "autoprefixer": "^9",
     "jasmine-core": "~2.4.1",
     "karma": "~6.3.4",
     "karma-chrome-launcher": "~3.1.0",
     "karma-coffee-preprocessor": "~1.0.1",
     "karma-jasmine": "~0.3.8",
+    "postcss": "^7",
```
Comment (Matt-Yorkley) [inline package.json:22]: Hmm I think we can actually use PostCSS 8 now, and drop the `@tailwindcss/postcss7-compat` :tada:

FYI this is where I would look for clues:
https://dev.to/davidteren/getting-started-with-rails-6-1-tailwindcss-jit-webpacker-postcss-8-25g7
https://github.com/davidteren/tails_on_rails

Comment (Matt-Yorkley) [reply package.json:22]: Or in this Gist: https://gist.github.com/leastbad/d4c7e1d7872a1e20c0e9a82340f2082f

Comment (jibees) [reply package.json:22]: Will recreate a PR ;)

**Pull request closed without merging**